samples/Ch10_expressing_kernels/fig_10_7_opencl_source_interop.cpp

```diff
@@ -22,14 +24,28 @@ int main() {
     // Note: This must select a device that supports interop!
     queue Q{ cpu_selector{} };
 
-    program p{Q.get_context()};
-    p.build_with_source(R"CLC(
+    cl_context OCLContext = get_native<backend::opencl>(Q.get_context());
+    cl_device_id OCLDeviceID = get_native<backend::opencl>(Q.get_device());
+
+    const std::string Source = R"CLC(
             kernel void add(global int* data) {
                 int index = get_global_id(0);
                 data[index] = data[index] + 1;
             }
-        )CLC",
-        "-cl-fast-relaxed-math");
+        )CLC";
+
+    const char *SourceStr = Source.c_str();
+    const size_t SourceSize = Source.size();
+
+    cl_int Ret;
+    cl_program OCLProgram =
+        clCreateProgramWithSource(OCLContext, 1, &SourceStr, &SourceSize, &Ret);
+    Ret = clBuildProgram(OCLProgram, 1, &OCLDeviceID, "-cl-fast-relaxed-math",
+                         nullptr, nullptr);
+    cl_kernel OCLKernel = clCreateKernel(OCLProgram, "add", &Ret);
+
+    kernel SYCLKernel =
+        make_kernel<backend::opencl>(OCLKernel, Q.get_context());
```

Comment:
After these chanegs this code sequence is basically identical to that in figure 10.8.

Since there is no equivalent to `build_with_source` in SYCL 2020 perhaps this sample should be removed instead?  We can keep a placeholder .cpp file with a comment to explain what happened.

Reply:
I agree.  Until a "from source" API is defined at the SYCL level (not clear if this will ever exist), we rely on backend "from source" APIs which are shown in the following figure.